service/history/task/cross_cluster_task.go

```diff
@@ -73,9 +76,35 @@ type (
 		taskProcessor   Processor       //TODO: wire up the dependency
 		redispatchFn    func(task Task) //TODO: wire up the dependency
 		maxRetryCount   dynamicconfig.IntPropertyFn
+		settable        future.Settable
 	}
 )
 
+// NewCrossClusterTaskForTargetCluster creates a CrossClusterTask
+// for the processing logic at target cluster
+// the returned the Future will be unblocked when after the task
+// is processed. The future value has type types.CrossClusterTaskResponse
+// and there will be not error returned for this future. All errors will
+// be recorded by the FailedCause field in the response.
+func NewCrossClusterTaskForTargetCluster(
```

Comment:
Seems like not used, did you need to add this in this PR?

Reply:
It's used at L222 in `cross_cluster_task_processor.go`. Not sure if you review that file, it's collapsed by default I think.

Reply:
What do you think to have the future return in all NewCrossClusterXXXTask() and no need for the NewCrossClusterTaskForTargetCluster?